proto/pulumi/provider.proto

```diff
@@ -215,32 +220,105 @@ message GetSchemaResponse {
+    // [](pulumirpc.ResourceProvider.Update) calls. If true, the provider should expect these fields to be populated in
+    // these calls.
+    bool sends_old_inputs = 5;
+
+    // True if and only if the caller supports sending old inputs and outputs as part of
+    // [](pulumirpc.ResourceProvider.Delete) calls. If true, the provider should expect these fields to be populated in
+    // these calls.
+    bool sends_old_inputs_to_delete = 6;
 }
 
+// `ConfigureResponse` is the type of responses sent by a [](pulumirpc.ResourceProvider.Configure) call. Its primary
+// purpose is to communicate features that the provider supports back to the caller.
 message ConfigureResponse {
-    bool acceptSecrets = 1;   // when true, the engine should pass secrets as strongly typed values to the provider.
-    bool supportsPreview = 2; // when true, the engine should invoke create and update with preview=true during previews.
-    bool acceptResources = 3; // when true, the engine should pass resources as strongly typed values to the provider.
-    bool acceptOutputs = 4;   // when true, the engine should pass output values to the provider.
+    // True if and only if the provider supports secrets. If true, the caller should pass secrets as strongly typed
+    // values to the provider.
+    bool acceptSecrets = 1;
+
+    // True if and only if the provider supports the `preview` field on [](pulumirpc.ResourceProvider.Create) and
+    // [](pulumirpc.ResourceProvider.Update) calls. If true, the engine should invoke these calls with `preview` set to
+    // `true` during previews.
+    bool supportsPreview = 2;
+
+    // True if and only if the provider supports strongly typed resources. If true, the caller should pass resources as
+    // strongly typed values to the provider.
+    bool acceptResources = 3;
+
+    // True if and only if the provider supports output values as inputs. If true, the engine should pass output values
+    // to the provider where possible.
+    bool acceptOutputs = 4;
 }
 
-// ConfigureErrorMissingKeys is sent as a Detail on an error returned from `ResourceProvider.Configure`.
+// `ConfigureErrorMissingKeys` is the type of error details that may be sent in response to a
+// [](pulumirpc.ResourceProvider.Configure) call when required configuration keys are missing.
 message ConfigureErrorMissingKeys {
+    // The type of key-value pairs representing keys that are missing from a [](pulumirpc.ResourceProvider.Configure)
+    // call.
     message MissingKey {
-        string name = 1;        // the Pulumi name (not the provider name!) of the missing config key.
-        string description = 2; // a description of the missing config key, as reported by the provider.
+        // The name of the missing configuration key.
+        //
+        // :::{note}
+        // This should be the *Pulumi name* of the missing key, and not any provider-internal or upstream name. Names
```

Comment:
How are nested properties to be handled here?

Reply:
Tracked in #17691